Based on f3466c988d67b90d8767c8173d9fae1b64211759 Mon Sep 17 00:00:00 2001
From: Thomas Loimer <thomas.loimer@tuwien.ac.at>
Date: Mon, 2 Sep 2024 21:57:04 +0200
Subject: [PATCH] With xfig -debug, do not draw bounding boxes into widgets

Do not draw bounding boxes into the example libraries and
example dimension line.
---
 src/u_draw.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git src/u_draw.c src/u_draw.c
index 71edbab..22740db 100644
--- src/u_draw.c
+++ src/u_draw.c
@@ -3,7 +3,7 @@
  * Copyright (c) 1985-1988 by Supoj Sutanthavibul
  * Parts Copyright (c) 1989-2015 by Brian V. Smith
  * Parts Copyright (c) 1991 by Paul King
- * Parts Copyright (c) 2016-2023 by Thomas Loimer
+ * Parts Copyright (c) 2016-2024 by Thomas Loimer
  *
  * Parts Copyright (c) 1992 by James Tough
  * Parts Copyright (c) 1998 by Georg Stemmer
@@ -1336,7 +1336,7 @@ void draw_text(F_text *text, int op)
 	return;
 
     /* outline the text bounds in red if debug resource is set */
-    if (appres.DEBUG) {
+    if (appres.DEBUG && !preview_in_progress) {
 	pw_vector(canvas_win, x1, y1, x2, y2, op, 1, RUBBER_LINE, 0.0, RED);
 	pw_vector(canvas_win, x2, y2, x3, y3, op, 1, RUBBER_LINE, 0.0, RED);
 	pw_vector(canvas_win, x3, y3, x4, y4, op, 1, RUBBER_LINE, 0.0, RED);
@@ -1593,7 +1593,7 @@ void clip_arrows(F_line *obj, int objtype, int op, int skip)
 		}
 		n = i;
 		/* draw the clipping area for debugging */
-		if (appres.DEBUG) {
+		if (appres.DEBUG && !preview_in_progress) {
 		  for (i=0; i<n; i++) {
 		    if (i==n-1)
 			j=0;
@@ -1632,7 +1632,7 @@ void clip_arrows(F_line *obj, int objtype, int op, int skip)
 		}
 		n = i;
 		/* draw the clipping area for debugging */
-		if (appres.DEBUG) {
+		if (appres.DEBUG && !preview_in_progress) {
 		  int j;
 		  for (i=0; i<n; i++) {
 		    if (i==n-1)
@@ -2127,7 +2127,7 @@ void debug_depth(int depth, int x, int y)
     char	str[10];
     PR_SIZE	size;
 
-    if (appres.DEBUG) {
+    if (appres.DEBUG && !preview_in_progress) {
 	sprintf(str,"%d",depth);
 	size = textsize(roman_font, strlen(str), str);
 	pw_text(canvas_win, x-size.length-round(3.0/zoomscale), round(y-3.0/zoomscale),
-- 
2.35.3

